crates/rethnet_eth/src/remote.rs

```diff
@@ -104,6 +119,17 @@ impl BlockSpec {
     }
 }
 
+impl Display for BlockSpec {
+    fn fmt(&self, formatter: &mut Formatter) -> Result<(), fmt::Error> {
+        let s = match self {
+            BlockSpec::Number(n) => n.to_string(),
+            BlockSpec::Tag(t) => t.to_string(),
+            BlockSpec::Eip1898(e) => serde_json::to_string(e).map_err(|_| fmt::Error)?,
```

Comment:
It felt appropriate since EIP-1898 specifies everything in JSON. `e.fmt(formatter)` would require `impl Display for Eip1898BlockSpec`, right? Do you think we should do a custom formatting besides just JSON? If so, what do you think that should look like?

Reply:
This was implied, but my comment requires that you implement `trait Display` for `Eip1898BlockSpec`.

My question specifically boils down to: Why is the struct itself not responsible for defining its output format?

This again comes down to single-responsibility principle: every unit is responsible for their own behaviour and it should be encapsulated within.

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/4095/commits/5ff2c6a9b46269161e97f44abd1016ecc4fbfd1b